server/queue/queue-activemq/src/main/java/org/apache/james/queue/activemq/ActiveMQMailQueueItem.java

```diff
@@ -52,7 +53,7 @@ public ActiveMQMailQueueItem(Mail mail, Session session, MessageConsumer consume
     public void done(boolean success) throws MailQueueException {
         super.done(success);
         if (success) {
-            if (message instanceof ActiveMQBlobMessage && getMail().getAttribute(JAMES_REUSE_BLOB_URL) == null) {
+            if (message instanceof ActiveMQBlobMessage && !AttributeUtils.getAttributeValueFromMail(getMail(), JAMES_REUSE_BLOB_URL).isPresent()) {
```

Comment:
Why not just `mail.getStronglyTypeAttribute(name).isPresent()` ?

Reply:
Because `getAttribute` returns an `Optional`.